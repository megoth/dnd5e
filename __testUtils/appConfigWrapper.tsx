import React, { ReactNode } from "react";
import AppConfigProvider from "../src/contexts/appConfig";
import mockAppConfig from "./mockAppConfig";

interface Props {
  children: ReactNode;
}

export default function AppConfigWrapper({ children }: Props) {
  const { errorsUrl, translationsUrl } = mockAppConfig();
  return (
    <AppConfigProvider errorsUrl={errorsUrl} translationsUrl={translationsUrl}>
      {children}
    </AppConfigProvider>
  );
}
