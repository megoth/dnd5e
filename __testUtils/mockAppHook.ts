import mockApp from "./mockApp";

export default function mockAppHook(mock: jest.Mock, app = mockApp()) {
  mock.mockReturnValue(app);
}
