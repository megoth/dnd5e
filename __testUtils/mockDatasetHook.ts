import { mockSolidDatasetFrom, SolidDataset } from "@inrupt/solid-client";
import { SWRResponse } from "swr";

export default function mockDatasetHook(
  mock,
  presets: Partial<SWRResponse<SolidDataset, any>> = {}
) {
  return mock.mockReturnValue({
    data: mockSolidDatasetFrom("http://example.com"),
    error: null,
    ...presets,
  });
}
