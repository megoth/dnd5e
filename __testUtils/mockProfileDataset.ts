import {
  addStringNoLocale,
  addUrl,
  mockSolidDatasetFrom,
  mockThingFrom,
  setThing,
} from "@inrupt/solid-client";
import { vcard, foaf, rdf } from "rdf-namespaces";
import { chain } from "../src/utils";

export const aliceWebIdDatasetUrl = "http://example.com/alice";
export const aliceWebIdUrl = `${aliceWebIdDatasetUrl}#me`;
export const aliceName = "Alice";
export const aliceNick = "A";
export const alicePhoto = "http://example.com/alice.jpg";

export function mockProfileDataset(webId = aliceWebIdUrl) {
  const profile = chain(
    mockThingFrom(webId),
    (t) => addStringNoLocale(t, vcard.fn, aliceName),
    (t) => addStringNoLocale(t, vcard.nickname, aliceNick),
    (t) => addUrl(t, vcard.hasPhoto, alicePhoto),
    (t) => addUrl(t, rdf.type, foaf.Person),
    (t) => addUrl(t, foaf.openid, aliceWebIdUrl)
  );
  return chain(mockSolidDatasetFrom(aliceWebIdDatasetUrl), (d) =>
    setThing(d, profile)
  );
}
