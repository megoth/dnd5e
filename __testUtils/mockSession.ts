import { Session } from "@inrupt/solid-client-authn-browser";

export interface SessionInfo {
  session: Session;
  sessionRequestInProgress: boolean;
  fetch: typeof window.fetch;
  login: () => Promise<void>;
  logout: () => Promise<void>;
}

export const authenticatedWebId = "http://example.com/webId#me";

export function mockAuthenticatedSession(options: Partial<Session> = {}) {
  return ({
    session: {
      info: {
        isLoggedIn: true,
        sessionId: "some-authenticated-session-id",
        webId: authenticatedWebId,
      },
      sessionRequestInProgress: false,
    },
    handleIncomingRedirect: () => Promise.resolve(),
    login: () => Promise.resolve(),
    logout: () => Promise.resolve(),
    on: () => {},
    ...options,
  } as unknown) as SessionInfo;
}

export function mockUnauthenticatedSession(options: Partial<Session> = {}) {
  return ({
    session: {
      info: {
        isLoggedIn: false,
        sessionId: "some-unauthenticated-session-id",
        webId: null,
      },
      sessionRequestInProgress: false,
    },
    handleIncomingRedirect: () => Promise.resolve(),
    login: options.login || jest.fn().mockResolvedValue(undefined),
    logout: () => Promise.resolve(),
    on: () => {},
    ...options,
  } as unknown) as SessionInfo;
}
