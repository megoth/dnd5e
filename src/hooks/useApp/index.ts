import { useContext, useEffect } from "react";
import { AppContext } from "../../contexts/app";
import { appIsLoadingLocalizations } from "../../models/app";

export default function useApp(bundles: string[] = []) {
  const { app, setBundles } = useContext(AppContext);
  useEffect(() => {
    if (!app || !appIsLoadingLocalizations(app, bundles)) {
      return;
    }
    setBundles(bundles);
  }, [app, bundles, setBundles]);
  return app;
}
