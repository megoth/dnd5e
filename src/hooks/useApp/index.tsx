import { useContext } from "react";
import AppContext from "./context";

export default function useApp() {
  return useContext(AppContext);
}
