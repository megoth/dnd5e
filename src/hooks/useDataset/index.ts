import useSWR from "swr";
import { getSolidDataset } from "@inrupt/solid-client";
import { useSession } from "@inrupt/solid-ui-react";

export default function useDataset(url) {
  const { fetch } = useSession();
  return useSWR([url, "dataset"], () => getSolidDataset(url, { fetch }));
}
