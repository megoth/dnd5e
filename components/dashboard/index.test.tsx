import { render } from "@testing-library/react";
import React from "react";
import { createRouter } from "next/router";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import useApp from "../../src/hooks/useApp";
import mockAppHook from "../../__testUtils/mockAppHook";
import Dashboard from "./index";
import useDataset from "../../src/hooks/useDataset";
import mockDatasetHook from "../../__testUtils/mockDatasetHook";
import { mockProfileDataset } from "../../__testUtils/mockProfileDataset";
import { authenticatedWebId } from "../../__testUtils/mockSession";

jest.mock("../../src/hooks/useApp");
const mockedAppHook = useApp as jest.Mock;

jest.mock("../../src/hooks/useDataset");
const mockedDatasetHook = useDataset as jest.Mock;

describe("Dashboard", () => {
  // @ts-ignore
  const router = createRouter("", {}, "", {});

  it("renders", () => {
    mockAppHook(mockedAppHook);
    mockDatasetHook(mockedDatasetHook, {
      data: mockProfileDataset(authenticatedWebId),
    });
    const { asFragment } = render(
      <RouterContext.Provider value={router}>
        <Dashboard />
      </RouterContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
