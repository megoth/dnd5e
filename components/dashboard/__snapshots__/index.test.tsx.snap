// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col relative min-h-full"
  >
    <header
      class="mb-2 shadow shadow-md"
    >
      <div
        class="main-container main-container--content pr-0"
      >
        <a
          class="px-4 lg:px-0 font-2xl font-serif self-center text-left focus:outline-none focus:ring-2 focus:ring-red-600"
          href="/"
        >
          [Translation for https://example.com/global/translations.ttl#appName does not exist]
        </a>
      </div>
      <div
        class="bg-gray-100 dark:bg-gray-900"
      >
        <div
          class="main-container main-container--content flex px-0"
        >
          <button
            class="p-2 px-4 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-red-600 lg:hidden"
            data-testid="page-header-left-menu-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="flex-1 text-lg font-semibold font-serif leading-10"
          >
            [Translation for https://example.com/global/translations.ttl#homePageTitle does not exist]
          </div>
          <button
            class="p-2 px-4 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-red-600"
            data-testid="page-header-right-menu-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 6V4m0 2a2 2 0 100 4m0-4a2 2 0 110 4m-6 8a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4m6 6v10m6-2a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </header>
    <div
      class="main-container"
    >
      <div
        class="layout__sidebar layout__sidebar--left  layout__sidebar--content"
        data-testid="layout-sub-menu"
      >
        <button
          class="button button--close layout__close-button layout__close-button--left layout__close-button--content text-right"
          data-testid="sub-menu-nav-close-button"
          type="button"
        >
          <svg
            class="icon"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          Â [Translation for https://example.com/global/translations.ttl#close does not exist]
        </button>
        <nav
          class="pages-nav pages-nav--sub-menu"
        >
          <ul
            class="nav-pages__list nav-pages__list--sub-menu"
          >
            <li>
              <a
                class="pages-nav__link"
                href="/characters"
              >
                [Translation for https://example.com/global/translations.ttl#charactersPageTitle does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                href="/rules"
              >
                [Translation for https://example.com/global/translations.ttl#rulesPageTitle does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                href="/faq"
              >
                [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
              </a>
            </li>
            <li>
              <a
                class="pages-nav__link"
                href="/about"
              >
                [Translation for https://example.com/global/translations.ttl#aboutPageTitle does not exist]
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="layout__main  layout__main--content"
      >
        <main
          class="flex-1"
        >
          <div
            class="content leading-normal max-w-prose mx-auto"
            style="hyphens: auto;"
          >
            <h1>
              You are logged in!
            </h1>
            <p>
              This part is in development, more will come.
            </p>
            <a
              href="/about"
            >
              Learn more about the project
            </a>
          </div>
          <p>
            [Translation for https://example.com/global/translations.ttl#loggedIn does not exist]
          </p>
          <button
            class="button w-full"
            data-testid="authenticated-logout-button"
            type="button"
          >
            [Translation for https://example.com/global/translations.ttl#logOut does not exist]
          </button>
        </main>
        <footer
          class="main-container main-container--content mt-8 text-center sm:text-left xs:flex xs:space-x-4 items-center justify-center sm:justify-start"
        >
          <div
            class="md:flex-grow pb-2 flex space-x-1 justify-center sm:justify-start"
          >
            <a
              class="link"
              href="/about"
            >
              [Translation for https://example.com/global/translations.ttl#aboutPageTitle does not exist]
            </a>
            <a
              class="link"
              href="/faq"
            >
              [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
            </a>
          </div>
          <div
            class="pb-2 flex xs:space-x-2 space-y-2 xs:space-y-0 flex-col text-center xs:flex-row xs:justify-center"
          >
            <button
              class="button"
              data-testid="dark-mode-selector-button"
              type="button"
            >
              <span>
                ðŸŒ™Â [Translation for https://example.com/global/translations.ttl#darkModeTurnOn does not exist]
              </span>
            </button>
          </div>
        </footer>
      </div>
      <div
        class="layout__sidebar layout__sidebar--right-fake  layout__sidebar--content"
      />
      <div
        class="layout__sidebar layout__sidebar--right  layout__sidebar--content"
      >
        <div>
          <button
            class="button button--close layout__close-button layout__close-button--right layout__close-button--content text-left"
            data-testid="session-close-button"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Â [Translation for https://example.com/global/translations.ttl#close does not exist]
          </button>
          <div
            class="px-2 xl:px-0"
          >
            <p>
              [Translation for https://example.com/global/translations.ttl#loggedIn does not exist]
            </p>
            <button
              class="button w-full"
              data-testid="authenticated-logout-button"
              type="button"
            >
              [Translation for https://example.com/global/translations.ttl#logOut does not exist]
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
